### PR TITLE
[project] Clean compile warnings.

### PR DESCRIPTION
**Pull request by harold**
- `lein check` is clean now
 - Added basic tests for untested areas I touched

---

`lein check` output on master:
```
$ lein check
Warning: implicit hook found: lein-npm.plugin/hooks 
Hooks are deprecated and will be removed in a future version.
Compiling namespace garden.def
Reflection warning, garden/color.cljc:183:25 - call to static method round on java.lang.Math can't be resolved (argument types: java.lang.Number).
Reflection warning, garden/color.cljc:398:30 - call to static method abs on java.lang.Math can't be resolved (argument types: unknown).
Reflection warning, garden/units.cljc:189:37 - reference to field getName can't be resolved.
Compiling namespace garden.types
Compiling namespace garden.units
Reflection warning, garden/units.cljc:189:37 - reference to field getName can't be resolved.
Compiling namespace garden.selectors
Compiling namespace garden.stylesheet
Compiling namespace garden.core
Compiling namespace garden.util
Compiling namespace garden.arithmetic
Compiling namespace garden.color
Reflection warning, garden/color.cljc:183:25 - call to static method round on java.lang.Math can't be resolved (argument types: java.lang.Number).
Reflection warning, garden/color.cljc:398:30 - call to static method abs on java.lang.Math can't be resolved (argument types: unknown).
Compiling namespace garden.compiler
Compiling namespace garden.repl
Reflection warning, garden/repl.clj:15:3 - call to method write can't be resolved (target class is unknown).
Reflection warning, garden/repl.clj:18:3 - call to method write can't be resolved (target class is unknown).
Reflection warning, garden/repl.clj:21:3 - call to method write can't be resolved (target class is unknown).
Reflection warning, garden/repl.clj:28:5 - call to method write can't be resolved (target class is unknown).
Reflection warning, garden/repl.clj:31:3 - call to method write can't be resolved (target class is unknown).
Compiling namespace garden.compression
Compiling namespace garden.media
Compiling namespace garden.def-test
Compiling namespace garden.arithmetic-test
Compiling namespace garden.compiler-test
WARNING: compile already refers to: #'clojure.core/compile in namespace: garden.compiler-test, being replaced by: #'garden.compiler-test/compile
Compiling namespace garden.selectors-test
Compiling namespace garden.util-test
Compiling namespace garden.tests
Compiling namespace garden.stylesheet-test
Compiling namespace garden.units-test
Compiling namespace garden.color-test
Compiling namespace user
Compiling namespace garden.def
Compiling namespace garden.types
Compiling namespace garden.units
Reflection warning, garden/units.cljc:189:37 - reference to field getName can't be resolved.
Compiling namespace garden.selectors
Compiling namespace garden.stylesheet
Compiling namespace garden.core
Compiling namespace garden.util
Compiling namespace garden.arithmetic
Compiling namespace garden.color
Reflection warning, garden/color.cljc:183:25 - call to static method round on java.lang.Math can't be resolved (argument types: java.lang.Number).
Reflection warning, garden/color.cljc:398:30 - call to static method abs on java.lang.Math can't be resolved (argument types: unknown).
Compiling namespace garden.compiler
Compiling namespace garden.repl
Reflection warning, garden/repl.clj:15:3 - call to method write can't be resolved (target class is unknown).
Reflection warning, garden/repl.clj:18:3 - call to method write can't be resolved (target class is unknown).
Reflection warning, garden/repl.clj:21:3 - call to method write can't be resolved (target class is unknown).
Reflection warning, garden/repl.clj:28:5 - call to method write can't be resolved (target class is unknown).
Reflection warning, garden/repl.clj:31:3 - call to method write can't be resolved (target class is unknown).
Compiling namespace garden.compression
Compiling namespace garden.media
```

---

`lein check` output on this branch:
```
$ lein check
Warning: implicit hook found: lein-npm.plugin/hooks 
Hooks are deprecated and will be removed in a future version.
Compiling namespace garden.def
Compiling namespace garden.types
Compiling namespace garden.units
Compiling namespace garden.selectors
Compiling namespace garden.stylesheet
Compiling namespace garden.core
Compiling namespace garden.util
Compiling namespace garden.arithmetic
Compiling namespace garden.color
Compiling namespace garden.compiler
Compiling namespace garden.repl
Compiling namespace garden.compression
Compiling namespace garden.media
Compiling namespace garden.def-test
Compiling namespace garden.arithmetic-test
Compiling namespace garden.compiler-test
Compiling namespace garden.selectors-test
Compiling namespace garden.util-test
Compiling namespace garden.tests
Compiling namespace garden.stylesheet-test
Compiling namespace garden.units-test
Compiling namespace garden.color-test
Compiling namespace user
Compiling namespace garden.def
Compiling namespace garden.types
Compiling namespace garden.units
Compiling namespace garden.selectors
Compiling namespace garden.stylesheet
Compiling namespace garden.core
Compiling namespace garden.util
Compiling namespace garden.arithmetic
Compiling namespace garden.color
Compiling namespace garden.compiler
Compiling namespace garden.repl
Compiling namespace garden.compression
Compiling namespace garden.media
```

### CODE REVIEW

**Comment by harold** [comment]
@noprompt - hi! :wave: 

Thanks for `garden` - we use it on lots of projects and if id didn't exist, we'd need to start by writing something just like it (but we'd do a worse job than you have).

These compile warnings bubble up into our tooling when we include this library, so I figured I'd dive in and fix them. I threw in a couple of trivial tests as well for untested functionality I touched.

Hope this helps and that you're safe and sound in these perplexing viral times.

Take care.

**Comment by noprompt** [comment]
@harold Thanks for this! Awesome! I did a quick review of the code and it looks good to me!

**Comment by noprompt** [comment]
@harold Pushed a new version of the library up. If there are any problems let me know. Also, I apologize that it took a moment to see this and review it. I've been very busy between family and work lately. The whole virus lockdown situation has sapped a lot of my open source energy but I will try and check in more periodically with the project.

**Comment by harold** [comment]
@noprompt - You're so welcome - I'm glad to help (I hope I didn't screw anything up, ha!)

And for sure no need to apologize for delay, these are perplexing times. In some ways the lockdown situation freed some time for me (to look at this and other open-source things besides). Some odd sense of balance, perhaps.

Regardless, take care, and stay rad.

**Comment by noprompt** [comment]
@harold I added you to the list of committers, you're welcome to contribute to the project however you like provided you still open issues and wait for review.

Somehow, I need to figure out how to manage this project and Meander effectively. We have the `alef` branch that @smahood has graciously contributed to that represents the "next" iteration of the library but I've fallen behind on getting it ready for public consumption.

In short, what I'm trying to say is if you're able to contribute and/or would like to, feel free to take action. 😄

**Comment by harold** [comment]
@noprompt - got it. The current version is serving us really well, but I imagine there are some cool improvements in the next.

I'll watch the repo and keep an eye out for anywhere I may be able to help. If you or @smahood or anyone else need anything, feel free to mention me in an issue, or w.e.

Thanks.
